frontend/src/app/containers/App.js

```diff
@@ -93,21 +98,72 @@ class App extends Component {
       locale: (navigator.language || '').split('-')[0]
     });
     this.props.chooseTests();
-    this.props.fetchUserCounts(config.usageCountsURL).then(() => {
+    const userCountsPromise = this.props.fetchUserCounts(config.usageCountsURL);
+    this.measurePageview();
+
+    const langs = {};
+
+    function addLang(lang, response) {
+      if (response.ok) {
+        return response.text().then(data => {
+          langs[lang] = `${langs[lang] || ''}${data}
+`;
+        });
+      }
+      return Promise.resolve();
+    }
+
+    const negotiated = negotiateLanguages(
+      navigator.languages,
+      config.AVAILABLE_LOCALES,
+      { defaultLocale: 'en-US' }
+    );
+
+    const promises = negotiated.map(language =>
+      Promise.all(
+        [
+          fetch(`/static/locales/${language}/app.ftl`).then(response => addLang(language, response)),
```

Comment:
If we do this, let's require() the languages.json file so that it is included in the bundle and doesn't add another round trip.